urllib3/exceptions.py

```diff
@@ -196,6 +196,14 @@ class ResponseNotChunked(ProtocolError, ValueError):
     pass
 
 
+class BodyNotHttplibCompatible(AssertionError, ValueError):
```

Comment:
HTTPError, to me, has a connotation that something happened during an actual connection (and most of the exceptions already in existence here confirm that). We can still use it as a base for most of our exceptions, but some of our exceptions don't inherit from that at all, so it's actually not our root exception and in my experience, users like having one last catch-all except block around a library (in case something funky goes wrong).

I think I was overzealous in asking HTTPWarning to inherit from Urllib3Exception. We don't raise that or it's sub-classes as exceptions so we can revert that.